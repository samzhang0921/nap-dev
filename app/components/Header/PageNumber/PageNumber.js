import React from 'react';
import {render} from 'react-dom';
import style from './PageNumber.css';
export default class PageNumber extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.changePage = this.changePage.bind(this);
  };
  changePage(page) {
    if (page < 0) {
      const npage = (this.props.currentPage === 1)
        ? 0
        : (this.props.currentPage-2) * 60;
      this.props.updateOffset(npage)
    } else if (page === 0) {
      const npage = (this.props.currentPage === this.props.totalPage)
        ? (this.props.totalPage - 1) * 60
        : this.props.currentPage * 60;
      this.props.updateOffset(npage)
    } else if (page > 0){
      this.props.updateOffset((this.props.page - 1) * 60)
    }
  }

  render() {
    const arrow = (this.props.page <= 0)
      ? ''
      : this.props.page;

    const dot = (isNaN(this.props.page))
      ? 'dot'
      : this.props.page;
    return (this.props.page === this.props.currentPage
      ? <li onClick={() => this.changePage(this.props.page)} className={style.pagenationBoxCurrentPage}>
          <span>{this.props.currentPage}</span>
        </li>
      : <li onClick={() => this.changePage(this.props.page)} className={[
        style['pagination-' + dot],
        style.pagenationBoxLi
      ].join(" ")}>
        <span>{arrow}</span>
      </li>)

  };

}
