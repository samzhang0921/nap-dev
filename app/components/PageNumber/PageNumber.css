.pagenationBoxLi {
margin-right: 10px;
align-self: auto;
cursor: pointer;
font-family: GillSans,"Gill Sans MT",Calibri,sans-serif;
font-size: 14px;
    transition: font-size 0.2s ease-in;
    -ms-transition: font-size 0.2s ease-in;
    -moz-transition: font-size 0.2s ease-in;
    -webkit-transition: font-size 0.2s ease-in;
}
.pagenationBoxCurrentPage {
margin-right: 5px;
align-self: auto;
font-family: GillSans,"Gill Sans MT",Calibri,sans-serif;

font-size: 16px;
transition: font-size 0.2s ease-in;
-ms-transition: font-size 0.2s ease-in;
-moz-transition: font-size 0.2s ease-in;
-webkit-transition: font-size 0.2s ease-in;
}
.pagination-dot {
  cursor: inherit;
}
.pagination--1 {
    border-width: 4.5px 7.8px 4.5px 0;
    border-color: transparent #000 transparent transparent;
    margin-bottom: 5px;
    border-style: solid;
}

.pagination-0 {
    border-width: 4.5px 0 4.5px 7.8px;
    border-color: transparent transparent transparent #000;
    margin-bottom: 5px;
    border-style: solid;
}
