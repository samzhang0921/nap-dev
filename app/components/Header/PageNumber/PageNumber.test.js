import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import PageNumber from './PageNumber';

describe('PageNumber', () => {
  const page = 10;
  const currentPage = 1;
  const totalPage = 15;
  const updateOffset = jest.fn();

      it('should mount without props ', () => {
      const tree = toJson(mount(<PageNumber />));
      expect(tree).toMatchSnapshot();
    });

    it('should match its snapshot with props', () => {

      const tree = renderer.create(
      <PageNumber page={page} currentPage={currentPage} totalPage={totalPage} updateOffset = {updateOffset} />
    ).toJSON();

      expect(tree).toMatchSnapshot();
    });

  it('should trigger updateOffset when its clicked', () => {
    const component = mount(<PageNumber page={page} currentPage={currentPage} totalPage={totalPage} updateOffset = {updateOffset} />);
    component.find('li').first().simulate('click');
      expect(updateOffset).toHaveBeenCalled();
  });
});
