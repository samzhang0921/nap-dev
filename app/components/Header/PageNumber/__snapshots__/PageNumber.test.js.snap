// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNumber should match its snapshot with props 1`] = `
<li
  className="pagination-10 pagenationBoxLi"
  onClick={[Function]}
>
  <span>
    10
  </span>
</li>
`;

exports[`PageNumber should mount without props  1`] = `
<PageNumber>
  <li
    className="pagenationBoxCurrentPage"
    onClick={[Function]}
  >
    <span />
  </li>
</PageNumber>
`;
